### DISCUSSION TITLE
called before a midi snapshot is sent

### DISCUSSION THREAD

**Discussion by jojogib** [RomanKubiak/ctrlr#518] [Q&A] [answered]
Is it possible to call more then 1  methode with "called before a midi snapshot is sent"?
thank you

**Comment by damiensellier** [comment, marked as answer]
Just call the many required functions within your "luaPanelMidiSnapshotPre"  single script,  it can cascade with as many methods as you need. Check out for loops if there's interconnections.

```
--
-- Called when data is restored
--
-- @buffer MidiBuffer all the midi messages that have been sent
--
myMethod = function(--[[ MidiBuffer --]] buffer)

myMethodSub1()
myMethodSub2()

end
```

And you need the 2x sub Methods in your LUA editor :

myMethodSub1 : 

```

function myMethodSub1()
	-- Your method code here
console("this is sub 1")
end
```

And myMethodSub2 :

```
function myMethodSub2()
	-- Your method code here
console("this is sub 2")
end
```

**Comment by jojogib** [comment]
Ok works great thank you :)
